src/Core/Managed/Shared/Extensibility/Implementation/SdkVersionUtils.cs

```diff
@@ -19,15 +20,20 @@ internal static string GetSdkVersion(string versionPrefix)
                     .OfType<AssemblyFileVersionAttribute>()
                     .First()
                     .Version;
-            
+
 #else
             string versionStr = typeof(TelemetryClient).GetTypeInfo().Assembly.GetCustomAttributes<AssemblyFileVersionAttribute>()
                     .First()
                     .Version;
 #endif
 
             Version version = new Version(versionStr);
-            return versionPrefix + version.ToString(3) + "-" + version.Revision;
+            string postfix = version.Revision.ToString(CultureInfo.InvariantCulture);
+#if NET40
+            postfix += "-fw4";
+#endif
+
+            return versionPrefix + version.ToString(3) + "-" + postfix;
```

Comment:
It might so happen that AI Endpoint relies on "-" to split SDK version into prefix and postfix, with this change, it will get "Version-Revision" as version (due to extra "-" from fw4) and might not apply necessary logic that depends on SDK version.